src/databricks/labs/ucx/azure/credentials.py

```diff
@@ -234,13 +255,33 @@ def _migrate_service_principals(
 
         execution_result = []
         for spn in sp_list_with_secret:
-            self._storage_credential_manager.create_with_client_secret(spn)
-            execution_result.append(self._storage_credential_manager.validate(spn.permission_mapping))
+            storage_credential_info = self._storage_credential_manager.create_with_client_secret(spn)
+            validation_results = self._storage_credential_manager.validate(
+                storage_credential_info,
+                spn.permission_mapping.prefix,
+            )
+            execution_result.append(validation_results)
         return execution_result
 
     def _create_access_connectors_for_storage_accounts(self) -> list[StorageCredentialValidationResult]:
-        self._resource_permissions.create_access_connectors_for_storage_accounts()
-        return []
+        access_connectors = self._resource_permissions.create_access_connectors_for_storage_accounts()
+
+        execution_results = []
+        for access_connector in access_connectors:
+            storage_credential_info = self._ws.storage_credentials.create(
+                access_connector.name,
+                azure_managed_identity=AzureManagedIdentity(str(access_connector.id)),
+                comment="Created by ucx",
+                read_only=False,
+            )
+            storage_account_name = access_connector.name.removeprefix("ac-")
+            validation_results = self._storage_credential_manager.validate(
+                storage_credential_info,
+                f"abfss://{storage_account_name}.dfs.core.windows.net/",
```

Comment:
@nfx : I am not sure if this is a valid url to validate the credential against. In the docs and our code, I only see urls with the `{container}@` in there. 

For the integration test, could we create an access connector with our test infra? So that it can be used in our test as we can't create Azure resources there.

Reply:
I've got some pending changes in our infra, once that is done, the access connector will be available as `TEST_ACCESS_CONNECTOR` env var, and that access connector will have blob contributor access to `TEST_MOUNT_CONTAINER`, which is the abfss path to the mount `TEST_MOUNT_NAME`

Reply:
`TEST_ACCESS_CONNECTOR` is available now

Reply:
@JCZuurmond @nkvuong `TEST_ACCESS_CONNECTOR` is available now

Reply:
@nkvuong or @nfx : Could I get read access on that access connector? I want to use it in my local test setup

Reply:
@JCZuurmond you should have read access on that access connector now